sycl/plugins/opencl/pi_opencl.cpp

```diff
@@ -342,6 +384,103 @@ pi_result piDeviceGetInfo(pi_device device, pi_device_info paramName,
     return PI_SUCCESS;
   }
 
+  case PI_DEVICE_INFO_PARTITION_PROPERTIES: {
+    // SYCL spec says: if this SYCL device cannot be partitioned into at least
+    // two sub devices then the returned vector must be empty.
+    pi_uint32 partitionMaxSubDevices = 0;
+    if (device->subLevel == -1)
+      device->subLevel = getSubLevel(device);
+    if (device->isRootDevice()) {
+      clGetDeviceInfo(
+          cast<cl_device_id>(device), CL_DEVICE_PARTITION_MAX_SUB_DEVICES,
```

Comment:
`cast<cl_device_id>(device)` - I'm confused here. It looks like the handle is from the OpenCL RT and is opaque for us, yet we do `device->subLevel` above.

Reply:
This handle (device) is a 'pi_device' handle. This way of passing the pi_device into OpenCL APIs seems like a standard way. I am not too sure about the 'internals' of OpenCL runtime. @bashbaug, can you please comment? Thanks